nums/core/array/utils.py

```diff
@@ -26,6 +27,60 @@
 # pylint: disable = no-member
 
 
+def find_slices(blocks):
+    src_params, dst_params, count = [], [], 0
+    for key in blocks:
+        start, end = blocks[key]
+        length = end - start
+        slice_obj_dst = slice(count, count + length, 1)
+        slice_obj_src = slice(start, end, 1)
+        src_params.append((slice_obj_src, False))
+        dst_params.append((slice_obj_dst, False))
+        count += length
+    return src_params, dst_params, count
+
+def find_output_blocks(X_blocks, out_shape, out_block_shape):
+    block_i, block_j, element_i, element_j = 0, 0, 0, 0
+    count, total_elements = 0, out_shape[0]
+    block, diagonal = X_blocks[(0, 0)], OrderedDict()
+    block_no, curr_block_length, blocks = 1, 0, []
```

Comment:
This is still difficult to read. Can you write these out on separate lines, and describe their roles briefly in the code? I understand what they do. We want to make it as easy as possible for future contributors to understand what's going on here.

Reply:
Also, change the name of blocks to `diag_meta` or something more interpretable. When I see a list named `blocks`, I think this is going to be a list of type `Block`.

Reply:
Will do!